python/taichi/lang/matrix.py

```diff
@@ -1436,8 +1436,11 @@ def __init__(self, n, m, ndim, dtype):
         #                 Remove the None dtype when we are ready to break legacy code.
         if dtype is not None:
             self.dtype = cook_dtype(dtype)
```

Comment:
Should reuse `self.dtype` to store the TensorType.

Reply:
But it seems that `MatrixType.dtype` is used in many places and it will take much time to rename it. Do you plan to remove `MatrixType` in the future? If so, I don't think we should waste time on this.

Reply:
MatrixType will be preserved since we'll need a type annotation for Matrix object anyway. I feel that having both `dtype` and `tensor_type` in the MatrixType is too confusing.

Reply:
Maybe we can refactor in another PR so that it will not block my successive PRs？

Reply:
+1 on having both `dtype` and `tensor_type` is confusing, @lin-hitonami if this PR is blocking some other PRs please feel free to merge first but a followup PR to eliminate `tensor_type` is pretty necessary to avoid making this part error prone. ;)